### PR TITLE
Feature: Support Intersection Observer

### PR DESCRIPTION
**Pull request by underbyte**
## Description

Adding support for `intersection observer` to nimbuild.

## Motivation and Context

* Devs are requesting this polyfill

## How Has This Been Tested?

* unit tests are updated

## Checklist:

-   [ ] My code follows the code style of this project.
-   [ ] My change requires a change to the documentation.
-   [ ] I have updated the documentation accordingly.
-   [ ] I have added tests to cover my changes.
-   [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/nimbuild-corejs/lib/module-resolver.js [hunk 1]
```diff
@@ -133,10 +151,14 @@ function getModules({features, logger, overrideTargetPlatform, uaString}) {
         modules.corejs = results.modules;
         targetPlatform = results.targetPlatform; // need to re-assign in case target platofrm fails in `getCoreJSModulesByPlatform()`
     }
-    if (!detectIfBrowserHasFetch(targetPlatform)) {
+    if (!detectIfBrowserHasFeature(targetPlatform, 'fetch')) {
         modules.normal.push('whatwg-fetch');
     }
 
+    if (!detectIfBrowserHasFeature(targetPlatform, 'intersectionobserver')) {
```
Comment (underbyte) [inline packages/nimbuild-corejs/lib/module-resolver.js]: @neenhouse `caniuse` seems to return `false` here regardless if the browser is modern or not.

Comment (underbyte) [reply packages/nimbuild-corejs/lib/module-resolver.js]: Found the issue. [caniuse](https://github.com/Fyrd/caniuse/blob/master/features-json/intersectionobserver.json#L188-L205) is reporting that in Chrome 65 and up it has a [side note](https://github.com/Fyrd/caniuse/blob/master/features-json/intersectionobserver.json#L376). So when `caniuse-api` assert if the feature was supported it only checks for "y" and not "y #4".

File: packages/nimbuild-corejs/package-lock.json [hunk 2]
```diff
@@ -494,19 +494,6 @@
             "integrity": "sha512-gCubfBUZ6KxzoibJ+SCUc/57Ms1jz5NjHe4+dI2krNmU5zCPAphyLJYyTOg06ueIyfj+SaCUqmzun7ImlxDcKg==",
             "dev": true
         },
-        "@vrbo/nimbuild-webpack": {
```
Comment (neenhouse) [inline packages/nimbuild-corejs/package-lock.json]: Did you intend to remove this?

Comment (underbyte) [reply packages/nimbuild-corejs/package-lock.json]: Weird. Not sure how that happened.

File: packages/nimbuild-corejs/lib/module-resolver.js (hunk 1 above)
Comment (underbyte) [reply packages/nimbuild-corejs/lib/module-resolver.js]: and now we wait https://github.com/Nyalab/caniuse-api/pull/90

File: packages/nimbuild-corejs/lib/index.js [hunk 3]
```diff
@@ -30,7 +34,10 @@ async function getPolyfillString({
     uaString
 }) {
     // Resolve base feature set based on include / exclude
-    const features = getBaseFeatureModules({include, exclude});
+    const features = getBaseFeatureModules({
```
Comment (neenhouse) [inline packages/nimbuild-corejs/lib/index.js]: Are these formatting changes driven by prettier?  When I edit this file, formatting snaps back to single line.

File: packages/nimbuild-corejs/lib/tests/index.test.js [hunk 4]
```diff
@@ -86,9 +99,8 @@ describe('index.js', () => {
             minify: true,
             logger: console
         });
-        console.log('INSPECT', polyfills);
+
         expect(polyfills.entry).toMatchSnapshot();
-        expect(polyfills.script.length).toBeGreaterThan(15000);
```
Comment (neenhouse) [inline packages/nimbuild-corejs/lib/tests/index.test.js]: Why was this assertion removed?

File: packages/nimbuild-corejs/lib/utils/isSupported.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
+const browserslist = require("browserslist");
+const { features, feature: readableFeature } = require("caniuse-lite");
+
+const featuresList = Object.keys(features);
```
Comment (neenhouse) [inline packages/nimbuild-corejs/lib/utils/isSupported.js]: Is this needed?  Seems to be unused.

File: packages/nimbuild-corejs/lib/utils/isSupported.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+ * @param {string} browsers - REQUIRED. Browsers to check the if it support the required feature param.
+ * @param {string} feature - REQUIRED. Browser feature to check if the browser support it.
+ * @returns {boolean}
+ */
+function isSupported(browsers, feature) {
+    let data;
+    try {
+        data = readableFeature(features[feature]);
+    } catch(e) {
+        throw new ReferenceError(`Please provide a proper feature name. Cannot find ${feature}`);
```
Comment (neenhouse) [inline packages/nimbuild-corejs/lib/utils/isSupported.js]: Line needs code coverage.

File: packages/nimbuild-corejs/lib/utils/isSupported.js [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+        return browserList.map((browser) => {
+            return browser.split(" ");
+        }).every((browser) => {
+            return data && data.stats[browser[0]] &&
+                data.stats[browser[0]][browser[1]] &&
+                String.prototype.toLowerCase.call(data.stats[browser[0]][browser[1]][0]) === "y"
+        });
+    }
+
+    throw new ReferenceError(`browser is an unknown version: ${browsers}`);
```
Comment (neenhouse) [inline packages/nimbuild-corejs/lib/utils/isSupported.js]: Line needs code coverage.

File: packages/nimbuild-corejs/package.json [hunk 8]
```diff
@@ -65,11 +65,12 @@
     ],
     "dependencies": {
         "@vrbo/nimbuild-webpack": "^0.0.3",
-        "browserslist": "^4.7.2",
+        "browserslist": "^4.8.5",
         "browserslist-useragent": "^3.0.2",
-        "caniuse-api": "^3.0.0",
```
Comment (neenhouse) [inline packages/nimbuild-corejs/package.json]: `1.0.30001023` !!!

**Pull request merged**